api/all/src/main/java/io/opentelemetry/api/logs/LogRecordBuilder.java

```diff
@@ -98,9 +103,110 @@ default LogRecordBuilder setAllAttributes(Attributes attributes) {
     return this;
   }
 
-  /** Sets an attribute. */
+  /**
+   * Sets an attribute to the newly created {@code LogRecord}. If {@code LogRecordBuilder}
+   * previously contained a mapping for the key, the old value is replaced by the specified value.
+   *
+   * <p>Note: the behavior of null values is undefined, and hence strongly discouraged.
+   *
+   * @param key the key for this attribute.
+   * @param value the value for this attribute.
+   * @return this.
+   */
   <T> LogRecordBuilder setAttribute(AttributeKey<T> key, T value);
 
+  /**
+   * Sets an attribute to the newly created {@code LogRecord}. If {@code LogRecordBuilder}
+   * previously contained a mapping for the key, the old value is replaced by the specified value.
+   *
+   * <p>Note: the behavior of null values is undefined, and hence strongly discouraged.
+   *
+   * @param key the key for this attribute.
+   * @param value the value for this attribute.
+   * @return this.
+   */
+  default <T> LogRecordBuilder setComplexAttribute(AttributeKey<T> key, ComplexAttribute value) {
```

Comment:
What about adding a method:
```
LogRecordBuilder setAttribute(String key, Value value);
```

Now that we have a specification definition for [standard attribute](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/common#standard-attribute) which perfectly aligns with our `Attribute` interface, I'd like to see if we can keep this alignment.

What do we loose if we provide log API setters for adding complex attributes, without any changes to `Attributes` / `AttributeKey`, `AttributeBuilder`?